### PR TITLE
[Backport stable/8.3] Avoid accidental reordering of column family values

### PR DESCRIPTION
**Pull request by korthout**
# Description
Backport of #16106 to `stable/8.3`.

Relates to #15804

Due to conflicts, I would recommend you review the commits individually. Commits that resolve conflicts mention this at the bottom of the commit message (under the `cherry picked from commit x` line).

There were conflicts because the following column family don't exist on 8.3 yet:
- `SIGNAL_SUBSCRIPTION_BY_NAME_AND_KEY(77)`
- `SIGNAL_SUBSCRIPTION_BY_KEY_AND_NAME(78)`
- `USER_TASKS(79)`
- `USER_TASK_STATES(80)`
- `COMPENSATION_SUBSCRIPTION(81)`

### CODE REVIEW

**Comment by korthout** [comment]
@nicpuppa @Zelldon I've asked for your review because you also reviewed the original.

Due to conflicts, I would recommend you review the commits individually. Commits that resolve conflicts mention this at the bottom of the commit message (under the cherry picked from commit x line).

**Review by nicpuppa — APPROVED** [review]
LGTM 🚀

File: protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java [hunk 1]
```diff
@@ -15,155 +15,165 @@
...
-  DMN_LATEST_DECISION_REQUIREMENTS_BY_ID,
-  DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY,
-  DMN_DECISION_KEY_BY_DECISION_ID_AND_VERSION,
-  DMN_DECISION_REQUIREMENTS_KEY_BY_DECISION_REQUIREMENT_ID_AND_VERSION,
+  JOB_ACTIVATABLE(76);
 
-  FORMS,
-  FORM_VERSION,
-  FORM_BY_ID_AND_VERSION,
+  private final int value;
```
Comment (Zelldon) [inline protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java:169]: 🔧 Same here I would expect to simply copy over the complete enum, this doesn't do any harm and is much simpler to backport in the future.

**Review by ChrisKujawa — APPROVED** [review]
See my comment below

File: protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java (hunk 1 above)
Comment (korthout) [reply protocol/src/main/java/io/camunda/zeebe/protocol/ZbColumnFamilies.java:169]: ✍️ For completeness, we discussed this here: https://github.com/camunda/zeebe/pull/16129#discussion_r1472809402

**Pull request merged**